experiments/experiment_prototype.py

```diff
@@ -1,385 +1,1111 @@
+                                                      'range {}'.format(freq_range))
+                        else:
+                            # TODO Log warning, changing clrfrqrange because lower portion is in a
+                            # restricted frequency range.
+                            exp_slice['clrfrqrange'][0] = freq_range[
+                                                              1] + 1  # outside of restricted range.
+                            break  # we have changed the 'clrfrqrange' - must restart the
+                            # check in case it's in another range.
+                    else:
+                        # lower end is not in restricted frequency range.
+                        try:
+                            assert exp_slice['clrfrqrange'][1] not in range(freq_range[0],
+                                                                            freq_range[1])
+                        except AssertionError:
+                            # TODO Log warning, changing clrfrqrange because upper portion is in a
+                            # restricted frequency range.
+                            exp_slice['clrfrqrange'][1] = freq_range[
+                                                              0] - 1  # outside of restricted range.
+                            break  # we have changed the 'clrfrqrange' - must restart the for loop
+                            # checking in case it's in another range.
+                        else:  # neither end of clrfrqrange is inside the restricted range but
+                            # we should check if the range is inside the clrfrqrange.
+                            try:
+                                assert freq_range[0] not in range(exp_slice['clrfrqrange'][0],
+                                                                  exp_slice['clrfrqrange'][1])
+                            except AssertionError:
+                                # TODO Log a warning that there is a restricted range in the middle
+                                # of the
+                                # clrfrqrange that will be avoided OR could make this an Error.
+                                # Still need to implement clear frequency searching.
+                                pass
+                else:  # no break, so no changes to the clrfrqrange
+                    still_checking = False
+
+        elif exp_slice['rxonly']:  # RX only mode.
+            # In this mode, rxfreq is required.
+            try:
+                assert isinstance(exp_slice['rxfreq'], int) or isinstance(exp_slice['rxfreq'],
+                                                                          float)
+                assert (exp_slice['rxfreq'] * 1000) < self.options.max_freq
+                assert (exp_slice['rxfreq'] * 1000) > self.options.min_freq
+            except AssertionError:
```

Comment:
we can, are we sure? I think to the user Hz resolution may look like overkill. ROS uses kHz.